### ISSUE TITLE
About the name of the community fork

### ISSUE THREAD

**Issue by mrnossiom** [tower-lsp-community/tower-lsp-server#3]
Having a distinct name resolves the following

- since there is little chance we can get the original crate ownership, we need to publish under something else than `tower-lsp`
- avoids confusing users. Having a distinct name makes it easy to know when the fork is used

Some proposition from the original issue (in time order):
- `tower-lsp2`
- `2tower-2lsp`
- `hightower-lsp`
- `tower-lzp` (like `exa` → `eza`)

**Comment by Decodetalkers** [comment]
towel-lsp? Lol

**Comment by trevor-scheer** [comment]
Thoughts:
* `tower-` follows common convention within the `tower` ecosystem for naming, sort were aim to preserve it
* `lsp` has meaning, so altering that would make it a lot less clear and discoverable
*  `2`s are confusing because of version numbers

So…I’m not a fan of any of our suggestions so far. Perhaps something matching`tower-lsp-*`? Not sure what to put there though.

**Comment by uncenter** [comment]
I'm a fan of `tower-lsp2`. Clearly a successor to the previous `tower-lsp`, though I can see the possible confusion with version numbers like you said.

**Comment by TilakMaddy** [comment]
It would be nice to have a sponsor so that this also doesn't end up like the original `tower-lsp`

**Comment by trevor-scheer** [comment]
@uncenter it is also my preference of the options so far.

Don't love but viable:
`tower-rs-lsp`
`tower-lsp-community`

**Comment by mrnossiom** [comment]
I agree with the points Trevor underlined earlier. And more than being confusing, I find the name `tower-lsp2` unpleasant. About `tower-rs-lsp`, I find that crates names containing `rs` are bad. I thought about `tower-lsp-community`, but it sounds like a bundle of community extensions to tower lsp.

The option I prefer is `tower-lsp2`

**Comment by TilakMaddy** [comment]
I am not convinced we should try so hard to have `tower` in the name of the fork

**Comment by TilakMaddy** [comment]
There are literally 100s of projects that are built on forks of one another and we have no idea, and I don't think that's a bad thing so long as github commits are not stolen and re-commited from some other name.

**Comment by trevor-scheer** [comment]
> I am not convinced we should try so hard to have `tower` in the name of the fork

@TilakMaddy I've provided a reason to keep it in the name, can you provide a reason not to? I'm open to the opinion if it's justified.

I'm most in favor of `tower-lsp2` for lack of a "best" option and would be very ok with that.

**Comment by Decodetalkers** [comment]
I am also in favor of tower-lsp2, seems this choice is good enough. I hope this project can be started as soon as possible..

**Comment by mrnossiom** [comment]
What about `tower-ls` (or `ls-tower` 👀) for making language servers with tower?

**Comment by Decodetalkers** [comment]
yes, tower-ls is greate!, it is a good name I think, ls can also means lsp. so, it is good

**Comment by trevor-scheer** [comment]
I like `tower-ls`. It's technically correct, this is an implementation and not the protocol itself.

Colloquially, people tend to refer to language server implementations as LSPs which is fine. I tend to lean into that when I'm talking about them since it's well understood. But I am comfortable going with correctness here since it gives us a name we aren't sad about and actual language server devs who use this package will likely understand the nuance.

**Comment by trevor-scheer** [comment]
One drawback: `ls` is very well known in a different way as the Unix command.

We could be more verbose:
`tower-languageserver`
`tower-language-server`

I like either of these as well. Crystal clear.

**Comment by Decodetalkers** [comment]
> One drawback: `ls` is very well known in a different way as the Unix command.
> 
> We could be more verbose: `tower-languageserver` `tower-language-server`
> 
> I like either of these as well. Crystal clear.

but tower-lanugageserver is too long, and think of the case of using macro, it seems a little inconvenient

**Comment by trevor-scheer** [comment]
Clarity over brevity IMO. The length of the package name is really not much of a concern for consumer projects.
Consumers wanting to switch to the fork will likely just want to update the one dependency line and alias it as `tower_lsp` right there. Can also alias on a per module basis if preferred.

That said, I don't dislike `tower-ls` but I do prefer `tower-language-server`.

**Comment by Decodetalkers** [comment]
have the name be decided? I want a release..

**Comment by mrnossiom** [comment]
It only annoys me because I would reserve language server (ls) to actual server implementations.

Let's just hope that no one will make a language called tower and try to make it's ls in Rust 🤞🏻.

If most people here are ok with that, I'll go with `tower-language-server`.

**Comment by mrnossiom** [comment]
Last proposal is `tower-lsp-server`. It has both `tower` and `lsp` keywords. It is close to `rust-analyzer`'s `lsp-server` name. Also, while being longer, it clearly indicates creating a server implementing the LSP protocol with `tower`.

I find that name reasonably long. This issue is blocking to release the crate and have a proper community fork.

I'll change the name to this in a week if most agree.
